### PR TITLE
Base Packages and Essentials Survey (July, 2024)

### PR DESCRIPTION
**Pull request by MingcongBai**
Topic Description
-----------------

- zstd: bump REL for topic Revision Marking Guidelines
- iw: bump REL for topic Revision Marking Guidelines
- dhcp: bump REL for topic Revision Marking Guidelines
- wpa-supplicant: bump REL for topic Revision Marking Guidelines
- aosc-aaa: bump REL for topic Revision Marking Guidelines
- zlib: bump REL for topic Revision Marking Guidelines
- gcc: bump REL for topic Revision Marking Guidelines
- wget: bump REL for topic Revision Marking Guidelines
- curl: bump REL for topic Revision Marking Guidelines
- xz: bump REL for topic Revision Marking Guidelines

... and 129 more commits

Package(s) Affected
-------------------

- admin-base: 14-1
- aosc-aaa: 11.5.0-2
- apt: 2:2.6.1-4
- bash: 5.2.26-2
- benchmark-base: 1-1
- boot-base: 5-1
- cinnamon-base: 2-1
- codec-base: 2-1
- coreutils: 9.5-2
- curl: 8.9.1-2
- dbus: 2:1.14.10-3
- debug-base: 3-1
- desktop-base: 12-1
- dhcp: 4.4.3.P1-2
- diffutils: 3.10-2
- dpkg: 1.22.7-2
- fcitx-base: 1-1
- fcitx5-base: 4-1
- findutils: 4.10.0-2
- gawk: 5.3.0-2
- gcc: 13.2.0-8
- gcc-runtime: 13.2.0-8
- gnome-base: 5-1
- grep: 3.11-2
- gzip: 1:1.13-2
- ibus-base: 1-1
- iproute2: 6.10.0-2
- iw: 5.19-2
- kbd: 2.5.1-2
- kernel-base: 4-1
- kmod: 31-2
- less: 1:643-3
- lxde-base: 2-1
- mate-base: 4-1
- nemo: 5.8.3-4
- net-tools: 1:2.10-3
- network-base: 6-1
- networkmanager: 1.40.6-3
- oma: 1.4.1-2
- print-base: 1-1
- procps: 4.0.4-2
- psmisc: 23.6-2
- python-base: 2-1
- retro-server-base: 1-1
- sed: 4.9-2
- server-base: 4-1
- shadow: 4.15.1-2
- sudo: 1.9.15p5-2
- tar: 1.35-2
- update-alternatives: 1.22.7-2
- util-base: 12-1
- util-linux: 2.40.1-2
- util-linux-runtime: 2.40.1-2
- web-base: 1-1
- wget: 1.24.5-2
- which: 2.21-5
- wpa-supplicant: 1:2.11-2
- xfce-base: 4-1
- xz: 5.6.2-2
- zlib: 1.3.1-3
- zstd: 1.5.6-2

Security Update?
----------------

No

Build Order
-----------

```
#buildit admin-base benchmark-base boot-base nemo cinnamon-base codec-base debug-base desktop-base fcitx5-base fcitx-base gnome-base ibus-base kernel-base lxde-base mate-base network-base print-base python-base retro-server-base server-base util-base web-base xfce-base apt dbus kmod oma shadow sudo iproute2 net-tools networkmanager bash coreutils diffutils findutils gawk grep gzip kbd less procps psmisc sed tar util-linux which xz curl wget gcc zlib aosc-aaa wpa-supplicant dhcp iw zstd
```

For architectures without GNOME (loongarch64, loongson3, ppc64el):

```
/build base-package-rework-survey-20240716 autobuild4,admin-base,benchmark-base,boot-base,nemo,cinnamon-base,codec-base,debug-base,desktop-base,fcitx5-base,fcitx-base,ibus-base,kernel-base,lxde-base,mate-base,network-base,print-base,python-base,retro-server-base,server-base,util-base,web-base,xfce-base,apt,dbus,kmod,oma,shadow,sudo,iproute2,net-tools,networkmanager,bash,coreutils,diffutils,findutils,gawk,grep,gzip,kbd,less,procps,psmisc,sed,tar,util-linux,which,xz,curl,wget,gcc,zlib,aosc-aaa,wpa-supplicant,dhcp,iw,zstd loongarch64,ppc64el,loongson3
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by cthbleachbit** [comment]
Stuff that probably shouldn't be ESSENTIAL:
- `inetutils`: curl can be used as an ftp client, `hostname` command is replaced by `hostnamectl` from systemd.
- `net-tools`: superseded by iproute2 in terms of function
- `patch`: probably won't be game-over if uninstalled

Stuff that probably should be ESSENTIAL:
- `wpa-supplicant` / `dhclient` / `iw`: core connectivity utilities used by NM or sysadmin manually
- `zstd` - AOSC OS ships zstd compressed kernel modules and firmware blobs. system might not survive after a reboot without zstd.

**Comment by KexyBiscuit** [comment]
Merging #7230 into this. Loop @MingcongBai

**Comment by MingcongBai** [comment]
> Stuff that probably shouldn't be ESSENTIAL:
> 
>     * `inetutils`: curl can be used as an ftp client, `hostname` command is replaced by `hostnamectl` from systemd.
> 
>     * `net-tools`: superseded by iproute2 in terms of function
> 
>     * `patch`: probably won't be game-over if uninstalled
> 
> 
> Stuff that probably should be ESSENTIAL:
> 
>     * `wpa-supplicant` / `dhclient` / `iw`: core connectivity utilities used by NM or sysadmin manually
> 
>     * `zstd` - AOSC OS ships zstd compressed kernel modules and firmware blobs. system might not survive after a reboot without zstd.

Addressed with the exception of `net-tools`. I for one still prefers `ifconfig`.

**Comment by KatyushaScarlet** [comment]
Stuff that probably shouldn't be ESSENTIAL in `server-base`:
* `lnmp-base` / `lamp-base` : For web servers, there are various environmental options available, such as Nginx/Apache/Caddy + MariaDB/PostgreSQL/... + (optional) PHP5/PHP7/PHP8 + .... Typically, a web server will select a specific combination that best suits its web application, compiling or upgrading the environment independently. Moreover, changing the version of software packages like PHP (upgrading from 5 to 7, or from 7 to 8) may break compatibility with the web application. I suggest not including `lnmp-base` / `lamp-base` in `server-base` because users' preferences vary, and choosing a default environment does not have universal applicability.

File: meta-bases/benchmark-base/spec [hunk 1]
```diff
@@ -1,3 +1,2 @@
-VER=0
-REL=1
+VER=2
```
Comment (KexyBiscuit) [inline meta-bases/benchmark-base/spec]: 2 or 1?

File: meta-bases/python-base/autobuild/defines [hunk 2]
```diff
@@ -1,6 +1,7 @@
 PKGNAME=python-base
 PKGSEC=Bases
-PKGDEP="python-2 python-3 pip virtualenv"
+PKGDEP="python-2 python-3"
```
Comment (KexyBiscuit) [inline meta-bases/python-base/autobuild/defines:3]: Python 2 until the finale.

**Review by KexyBiscuit — DISMISSED** [review]
LGTM overall.

**Review by xtexx — APPROVED** [review]

**Pull request merged**